packages/core/src/binary.rs

```diff
@@ -67,14 +56,14 @@ impl Binary {
     /// Copy to integer
     ///
     /// ```
-    /// # use cosmwasm_std::Binary;
+    /// # use cosmwasm_core::Binary;
     /// let binary = Binary::from(&[0x8b, 0x67, 0x64, 0x84, 0xb5, 0xfb, 0x1f, 0x37]);
     /// let num = u64::from_be_bytes(binary.to_array().unwrap());
     /// assert_eq!(num, 10045108015024774967);
     /// ```
-    pub fn to_array<const LENGTH: usize>(&self) -> StdResult<[u8; LENGTH]> {
+    pub fn to_array<const LENGTH: usize>(&self) -> CoreResult<[u8; LENGTH]> {
```

Comment:
Would it make sense/is it possible to move `StdResult`/`StdError` to core? Otherwise this is clearly breaking users

Reply:
Will have a look. I *don't think* it has any dependencies on `std`-only code

Reply:
Just gotta check if `cosmwasm_crypto` is already "core-ified" (because of the verification and key recovery structs having dependencies on `cosmwasm_crypto`)

Reply:
Mmm.. it technically *would* compile (as far as I can get it to compile at least), but the `getrandom` dependency just really doesn't like bare-metal..  

Well, that's a lie, there is a feature on `getrandom` that allows a fallback to `rdrand` which is a randomness hardware intrinsic of x86 hardware. If we enable that fallback in `getrandom` and test against `x86_64-unknown-none`, it could work just fine.

Opinions?

Reply:
*Or*, since we apparently don't shy away from breaking semver inside the crypto crate, we can simply add an `rng` parameter to all the functions requiring randomness, making the crate compile on all kinds of `no_std` targets.  
Something like `rng: &mut R where R: CryptoRng + RngCore` should be enough

Reply:
Oh nice, we don't even sign with ECDSA, so we don't even need randomness (well, not like we actually need it with RFC6979), so we just need it for Ed25519 batch verification.

Reply:
Okay, everything moved now.